zerver/migrations/0143_reactions_realm_emoji_name_to_id.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+
+import ujson
+
+from collections import defaultdict
+from django.conf import settings
+from django.db import migrations, models
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from typing import Any, Dict
+
+def realm_emoji_name_to_id(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    Reaction = apps.get_model('zerver', 'Reaction')
+    RealmEmoji = apps.get_model('zerver', 'RealmEmoji')
+    realm_emoji_by_realm_id = defaultdict(dict)   # type: Dict[int, Dict[str, Any]]
+    for realm_emoji in RealmEmoji.objects.all():
+        realm_emoji_by_realm_id[realm_emoji.realm_id] = {
+            'id': str(realm_emoji.id),
+            'name': realm_emoji.name,
+            'deactivated': realm_emoji.deactivated,
+        }
+    for reaction in Reaction.objects.filter(reaction_type='realm_emoji'):
+        realm_id = reaction.user_profile.realm_id
+        emoji_name = reaction.emoji_name
+        if realm_id in realm_emoji_by_realm_id:
```

Comment:
What about the `else` case for this `if`? We'll just leave the reaction lying around with a now-corrupted `emoji_code`.

Pretty sure it should be treated the same as the `realm_emoji is None` case below.

Reply:
Pushed a fix for this as 4139e6c76.